Added TSheets adapter

kandadaboggu opened the pull request:
- Adds Query Timesheets action to TSheets

eeshan92 commented on custom_auth/tsheets_connector.rb:
```diff
@@ -0,0 +1,89 @@
+  },
+
+  test: lambda do
+    true
```
Don't think we should encourage developers from forcing a connection success since they are rarely in charge of recipe building. Actual recipe builder might not be aware and provide invalid token.
We can use: `https://rest.tsheets.com/api/v1/current_user`

eeshan92 commented on custom_auth/tsheets_connector.rb:
```diff
@@ -0,0 +1,89 @@
+    authorization: {
+      type: "custom_auth",
+
+      credentials: lambda do
```
missing connection params here

eeshan92 commented on custom_auth/tsheets_connector.rb:
```diff
@@ -0,0 +1,89 @@
+      {
+        name: "api_token",
+        label: "API token",
+        optional: false,
```
make field `type: :password`

kandadaboggu commented:
@eeshan92 Addressed all the comments. This action can be refined to include other filters https://developers.tsheets.com/docs/api/timesheets/list-timesheets
